examples/threading/threading.c

```diff
@@ -14,6 +21,31 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    struct thread_data* thread_func_args = (struct thread_data*)thread_param;
+
+    // Sleep for the specified time before obtaining the mutex
+    usleep(thread_func_args->wait_to_obtain_ms * 1000);
+
+    // Obtain the mutex
+    if (pthread_mutex_lock(thread_func_args->mutex) != 0) {
+        thread_func_args->thread_complete_success = false;
+        pthread_exit(NULL);
+    }
+
+    // Sleep for the specified time while holding the mutex
+    usleep(thread_func_args->wait_to_release_ms * 1000);
+
+    // Release the mutex
+    if (pthread_mutex_unlock(thread_func_args->mutex) != 0) {
+        thread_func_args->thread_complete_success = false;
+        pthread_exit(NULL);
```

Comment:
In this path is the memory cleaning handled properly? what happens to thread_param